scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -192,6 +193,8 @@ sealed trait Execution[+T] extends java.io.Serializable {
    */
   def zip[U](that: Execution[U]): Execution[(T, U)] =
     Zipped(this, that)
+
+  override val hashCode = super.hashCode
```

Comment:
@jnievelt any time we have call-by-name, equality is not going to work, but we only use call-by-name in a couple of places. In your example, we are just forcing the `Try` which, if we are going to do that, why make it call by name?

The idea was that call-by-name is expensive in general, but in `Execution` the real cost is running data flows, and call-by-name is nothing compared to that.

That said, it does not kill the caching. Consider:

```scala
val a = loadStart()
val b = a.flatMap(foo)
val c = a.flatMap(bar)
val d = b.zip(c)
```
in this case, `a` will only be executed once. I realize this case also works with the `identityHashCode`, but I still think we should seek to maximize where the cache can work: which means, yes any function is a black box that uses reference equality, but there are many non-function cases.